### PR TITLE
Added a 'fling' gesture to WPImageViewController.

### PR DESCRIPTION
**Pull request by frosty**
Adds a 'fling' to dismiss gesture to `WPImageViewController`, like Slack and Tweetbot. 

### To test

* View an image full screen anywhere that `WPImageViewController` is used. Images in the Reader are probably easiest.
* Check that you can still zoom and pan the image. 
* When you're zoomed out you should be able to drag the image. If you fling it hard enough, it'll fly offscreen and be dismissed.
* Check when the app is viewed at different sizes. I changed the velocity slightly at regular vs compact sizes, as it felt a little fast when viewed Regular.
* Let me know if you think it 'feels' right! Compare to Slack and Tweetbot perhaps.

Needs review: @jleandroperez

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/WPImageViewController.m [hunk 1]
```diff
@@ -256,4 +264,22 @@ + (BOOL)isUrlSupported:(NSURL *)url
...
 }
 
+#pragma mark - FlingableViewHandlerDelegate
+
+- (void)flingableViewHandlerDidBeginRecognizingGesture:(FlingableViewHandler *)handler
+{
+    self.scrollView.multipleTouchEnabled = NO;
+}
+
+- (void)flingableViewHandlerDidEndRecognizingGesture:(FlingableViewHandler *)handler {
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/WPImageViewController.m]: Yup. You were right. Looks like an Apple API!

File: WordPress/Classes/ViewRelated/System/FlingableViewHandler.swift [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            break
+        case .Changed:
+            let anchor = recognizer.locationInView(referenceView)
+            attachmentBehavior.anchorPoint = anchor
+            break
+        case .Ended, .Cancelled:
+            animator.removeAllBehaviors()
+
+            let magnitude = recognizer.magnitudeInView(referenceView)
+            if magnitude < escapeVelocity {
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/System/FlingableViewHandler.swift]: Absolutely beautiful code @frosty!!

File: WordPress/Classes/ViewRelated/System/FlingableViewHandler.swift [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            break
+        default:
+            break
+        }
+    }
+
+    private func makeAttachmentBehaviorForRecognizer(recognizer: UIGestureRecognizer, inView view: UIView) -> UIAttachmentBehavior {
+        let location = recognizer.locationInView(view)
+
+        let (centerX, centerY) = (view.bounds.width / 2.0, view.bounds.height / 2.0)
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/System/FlingableViewHandler.swift]: Maybe *round* so that center never contains positions with comma?

**Comment by jleandroperez** [comment]
@frosty sorry to say this doesn't really *feels right*...

It's **PERFECT**.

One minor `round()` comment added, still, not sure if that actually affects anything, since the image never gets blurry. Tested on a 6s / iPad (in all of the multitasking sizes)

:shipit: sir!!!

**Comment by frosty** [comment]
Thank you for an amazing review, @jleandroperez!

**Pull request merged**